src/container/internal.tsx

```diff
@@ -109,50 +116,65 @@ export default function InternalContainer({
         styles.root,
         styles[`variant-${variant}`],
         fitHeight && styles['fit-height'],
+        media?.content && (mediaPosition === 'top' ? styles['with-top-media'] : styles['with-side-media']),
         shouldHaveStickyStyles && [styles['sticky-enabled']]
       )}
       ref={mergedRef}
     >
-      {header && (
-        <StickyHeaderContext.Provider value={{ isStuck }}>
-          <div
-            className={clsx(styles.header, styles[`header-variant-${variant}`], {
-              [styles['header-sticky-disabled']]: __stickyHeader && !isSticky,
-              [styles['header-sticky-enabled']]: isSticky,
-              [styles['header-dynamic-height']]: hasDynamicHeight,
-              [styles['header-stuck']]: isStuck,
-              [styles['with-paddings']]: !disableHeaderPaddings,
-              [styles['with-hidden-content']]: !children || __hiddenContent,
-            })}
-            {...headerIdProp}
-            {...stickyStyles}
-            ref={headerMergedRef}
-          >
-            {__darkHeader ? (
-              <div className={clsx(styles['dark-header'], 'awsui-context-content-header')}>{header}</div>
-            ) : (
-              header
-            )}
-          </div>
-        </StickyHeaderContext.Provider>
+      {media?.content && (
+        <div className={clsx(styles[`media-${mediaPosition}`], styles.media)} style={getMediaStyles()}>
+          {mediaContent}
+        </div>
       )}
       <div
-        className={clsx(styles.content, {
-          [styles['with-paddings']]: !disableContentPaddings,
-        })}
+        className={clsx(
+          styles['content-wrapper'],
+          media?.content && styles['content-wrapper-with-media'],
+          fitHeight && styles['content-wrapper-fit-height']
+        )}
       >
-        {children}
-      </div>
-      {footer && (
+        {header && (
+          <StickyHeaderContext.Provider value={{ isStuck }}>
```

Comment:
So what happens if a sticky header has media in it? 🤔

Reply:
This combination should not be possible for consumers of the Container.

`__stickyHeader` is a private property that we only use internally for having the sticky container throughout other components (like Cards or Table). I don't see us wanting both to coexist in these scenarios, so I think it is safe to ignore the corner case.